libs/payments/ui/src/lib/client/components/PaymentMethodManagement/index.tsx

```diff
@@ -0,0 +1,228 @@
+import { LoadingSpinner } from '../LoadingSpinner';
+import { useState, useRef } from 'react';
+import { StripePaymentElementChangeEvent } from '@stripe/stripe-js';
+import { Localized, useLocalization } from '@fluent/react';
+import { useRouter } from 'next/navigation';
+
+export function PaymentMethodManagement({
+  onSubmit,
+  defaultPaymentMethodId,
+}: {
+  onSubmit: (confirmationTokenId: string) => any;
+  defaultPaymentMethodId: string;
+}) {
+  const { l10n } = useLocalization();
+  const stripe = useStripe();
+  const elements = useElements();
+  const formRef = useRef<HTMLFormElement>(null);
+  const router = useRouter();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isComplete, setIsComplete] = useState(false);
+  const [isReady, setIsReady] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [isInputNewCardDetails, setIsInputNewCardDetails] = useState(false);
+  const [fullName, setFullName] = useState('');
+  const [hasFullNameError, setHasFullNameError] = useState(false);
+  const [isNonDefaultCardSelected, setIsNonDefaultCardSelected] =
+    useState(false);
+
+  const handleReady = () => {
+    setIsReady(true);
+  };
+
+  const handlePaymentElementChange = (
+    event: StripePaymentElementChangeEvent
+  ) => {
+    console.log('event', event);
+    setIsComplete(event.complete);
+
+    if (event.value.type === 'card' && !event.value.payment_method) {
+      setIsInputNewCardDetails(true);
+
+      if (event.complete) {
+        setHasFullNameError(fullName.length === 0);
+      }
+    } else if (event.value.type === 'card' && !!event.value.payment_method) {
+      setIsInputNewCardDetails(false);
+
+      if (event.value.payment_method.id !== defaultPaymentMethodId) {
+        setIsNonDefaultCardSelected(true);
+      } else {
+        setIsNonDefaultCardSelected(false);
+      }
+    }
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!stripe || !elements || !isComplete) {
+      return;
+    }
+
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      elements.submit();
+      const { confirmationToken, error: confirmationTokenError } =
+        await stripe.createConfirmationToken({
+          elements,
+          params: {
+            payment_method_data: { billing_details: { name: fullName } },
+          },
+        });
+
+      if (confirmationTokenError) {
+        throw confirmationTokenError;
+      }
+
+      await onSubmit(confirmationToken.id);
+      router.refresh();
+    } catch (error) {
+      console.error(error);
+      setError(error.message || 'An unexpected error occurred.');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <div className="p-4">
+      {!isReady && (
+        <div className="absolute inset-0 bg-white bg-opacity-75 flex items-center justify-center z-10">
+          <LoadingSpinner />
+        </div>
+      )}
```

Comment:
issue: The `<LoadingSpinner />` seems to clash with the payment element loading state, resulting in an awkward visual during loading.

suggestion: The payment element already provides a loading state. Could this change be removed in favor of using that?